### PR TITLE
Remove some duplicate routes, tighten up some route regexes in routes.go

### PR DESCRIPTION
**Pull request by rawlinp**
## What does this PR (Pull Request) do?

Remove some duplicate routes, add a unit test to check for duplicate route problems in the future, and tighten up some rewritten route regexes to match their corresponding Perl routes.

- [x] This PR is not related to any Issue

## Which Traffic Control components are affected by this PR?

- Traffic Ops

Fixes some API routes, no docs necessary.

## What is the best way to verify this PR?
Run the TO unit and API tests, verify they pass. Manually query some of the fixed routes to confirm they only accept literal `.config` instead of `<insert any character here>config` and `deliveryservices` instead of either `deliveryservice` or `deliveryservices` (since the original `?` in those routes made the `'s'` optional).

## If this is a bug fix, what versions of Traffic Control are affected?
- master

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/routing/routing_test.go [hunk 1]
```diff
@@ -174,11 +174,18 @@ func TestCompileRoutes(t *testing.T) {
 
 func TestRoutes(t *testing.T) {
 	fake := ServerData{Config: config.NewFakeConfig()}
-	_, _, _, err := Routes(fake)
+	routes, _, _, err := Routes(fake)
 	if err != nil {
 		t.Fatalf("expected: no error getting Routes, actual: %v", err)
 	}
 	// TODO: verify that all returned Routes are unique
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/routing/routing_test.go]: You wanna get rid of this TODO?

File: traffic_ops/traffic_ops_golang/routing/routes.go [hunk 2]
```diff
@@ -511,17 +508,17 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
...
 
 		// Federation Resolvers
 		{1.1, http.MethodPost, `federation_resolvers(/|\.json)?$`, federation_resolvers.Create, auth.PrivLevelAdmin, Authenticated, nil, 1134373661, perlBypass},
 		{1.1, http.MethodGet, `federation_resolvers(/|\.json)?$`, federation_resolvers.Read, auth.PrivLevelReadOnly, Authenticated, nil, 556608759, perlBypass},
 
 		// Federations Users
-		{1.1, http.MethodPost, `federations/{id}/users?(\.json)?$`, federations.PostUsers, auth.PrivLevelAdmin, Authenticated, nil, 1779334930, perlBypass},
-		{1.1, http.MethodGet, `federations/{id}/users?(\.json)?$`, api.ReadHandler(&federations.TOUsers{}), auth.PrivLevelReadOnly, Authenticated, nil, 394075015, perlBypass},
+		{1.1, http.MethodPost, `federations/{id}/users(\.json)?$`, federations.PostUsers, auth.PrivLevelAdmin, Authenticated, nil, 1779334930, perlBypass},
+		{1.1, http.MethodGet, `federations/{id}/users(\.json)?$`, api.ReadHandler(&federations.TOUsers{}), auth.PrivLevelReadOnly, Authenticated, nil, 394075015, perlBypass},
```
Comment (ocket8888) [inline traffic_ops/traffic_ops_golang/routing/routes.go]: I think the ones with the odd floating <kbd>?</kbd> were meant to be `{etc.}/?(\.json)?$` since the Perl API allowed trailing `/` and trailing `.json`

File: traffic_ops/traffic_ops_golang/routing/routing_test.go (hunk 1 above)
Comment (rawlinp) [reply traffic_ops/traffic_ops_golang/routing/routing_test.go]: yes, I do :)

File: traffic_ops/traffic_ops_golang/routing/routes.go (hunk 2 above)
Comment (rawlinp) [reply traffic_ops/traffic_ops_golang/routing/routes.go]: yeah, probably. I considered that, but we should probably allow an optional trailing `/` like everything else. I'll fix that.
